File: Cmdlets/CurrencyRegions.vb
Public Class CurrencyRegions

    Private _Regions As System.Collections.Generic.List(Of System.Globalization.RegionInfo)

    Public Sub New()
        Me._Regions = New System.Collections.Generic.List(Of System.Globalization.RegionInfo)
    End Sub

    Public Property Currency() As TIKSN.Finance.CurrencyInfo

    Public ReadOnly Property Regions() As System.Collections.Generic.List(Of System.Globalization.RegionInfo)
        Get
            Return Me._Regions
        End Get
    End Property

    Public ReadOnly Property Count() As Integer
        Get
            Return Me.Regions.Count
        End Get
    End Property
End Class

File: Common/Models/CurrencyRegions.cs
namespace TIKSN.Cmdlets.Common.Models
{
    public class CurrencyRegions
    {
        private System.Collections.Generic.List<System.Globalization.RegionInfo> _Regions;

        public CurrencyRegions()
        {
            this._Regions = new System.Collections.Generic.List<System.Globalization.RegionInfo>();
        }

        public int Count
        {
            get { return this.Regions.Count; }
        }

        public TIKSN.Finance.CurrencyInfo Currency { get; set; }

        public System.Collections.Generic.List<System.Globalization.RegionInfo> Regions
        {
            get { return this._Regions; }
        }
    }
}